test/e2e/rollout_test.go

```diff
@@ -542,6 +546,112 @@ var _ = Describe("placing wrapped resources using a CRP", Ordered, func() {
+				},
+			}
+		})
+
+		It("create the resources", func() {
+			Expect(hubClient.Create(ctx, &workNamespace)).To(Succeed(), "Failed to create namespace %s", workNamespace.Name)
+			Expect(hubClient.Create(ctx, &testCustomResource)).To(Succeed(), "Failed to create test custom resource %s", testCustomResource.GetName())
+		})
+
+		It("create the CRP that select the namespace and CRD", func() {
+			crp = buildCRPForSafeRollout()
+			crdClusterResourceSelector := placementv1beta1.ClusterResourceSelector{
+				Group:   utils.CRDMetaGVK.Group,
+				Kind:    utils.CRDMetaGVK.Kind,
+				Version: utils.CRDMetaGVK.Version,
+				Name:    testResourceCRDName,
+			}
+			crp.Spec.ResourceSelectors = append(crp.Spec.ResourceSelectors, crdClusterResourceSelector)
+			crp.Spec.Policy = &placementv1beta1.PlacementPolicy{
+				PlacementType: placementv1beta1.PickFixedPlacementType,
+				ClusterNames: []string{
+					memberCluster1EastProdName,
+				},
+			}
+			crp.Spec.Strategy.RollingUpdate.UnavailablePeriodSeconds = ptr.To(60)
+			Expect(hubClient.Create(ctx, crp)).To(Succeed(), "Failed to create CRP")
+		})
+
+		It("should update CRP status as expected", func() {
+			crpStatusUpdatedActual := customizedCRPStatusUpdatedActual(crpName, wantSelectedResources, []string{memberCluster1EastProdName}, nil, "0", false)
+			Eventually(crpStatusUpdatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update CRP status as expected")
+		})
+
+		It("should place the resources on member cluster", func() {
+			workResourcesPlacedActual := waitForTestResourceToBePlaced(memberCluster1EastProd, &testCustomResource)
+			Eventually(workResourcesPlacedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to place work resources on member cluster %s", memberCluster1EastProd.ClusterName)
+		})
+
+		It("update the custom resource", func() {
+			Eventually(func() error {
+				var cr testv1alpha1.TestResource
+				err := hubClient.Get(ctx, types.NamespacedName{Name: testCustomResource.Name, Namespace: workNamespace.Name}, &cr)
+				if err != nil {
+					return err
+				}
+				cr.Spec.Foo = "bar1" // Previously was "foo1"
+				return hubClient.Update(ctx, &cr)
+			}, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update custom resource")
+		})
+
+		It("should not update the resource on member cluster", func() {
+			// subtracting 5 seconds because transition between IT takes ~1 second
+			unavailablePeriod := time.Duration(*crp.Spec.Strategy.RollingUpdate.UnavailablePeriodSeconds)*time.Second - (5 * time.Second)
+			Consistently(func() error {
+				var cr testv1alpha1.TestResource
+				err := memberCluster1EastProd.KubeClient.Get(ctx, types.NamespacedName{Name: testCustomResource.Name, Namespace: workNamespace.Name}, &cr)
+				if err != nil {
+					return err
+				}
+				if cr.Spec.Foo != "bar1" { // Previously was "foo1"
+					return nil
+				}
+				return err // Other errors should be returned
```

Comment:
wouldn't this alwasy return nil, thus always pass?

Reply:
you're right.